build/config.props

```diff
@@ -26,7 +26,7 @@
     <!-- Visual Studio Insertion Logic -->
     <VsTargetMajorVersion>$([MSBuild]::Add(11, $(MajorNuGetVersion)))</VsTargetMajorVersion>
     <VsTargetBranch>main</VsTargetBranch>
-    <VsTargetChannel>int.$(VsTargetBranch)</VsTargetChannel>
+    <VsTargetChannel>int.d17.3</VsTargetChannel>
```

Comment:
Isn't this used by opt prof as well? It was originally added for opt prof purposes.

Reply:
Good question Nikolche. `VsTargetChannel` is used to generate Visual studio bootstrapper https://github.com/NuGet/NuGet.Client/blob/451713c59f5206f77c97d093dd5191be13f9d0e0/eng/pipelines/templates/Build_and_UnitTest.yml#L303 also.This bootstrapper is used in E2E, Apex and OptProf tests. One thing specific about OptProf is that as you mentioned it generates profiling data.

I compared the bootstrapper generated for this private build and an last official build. I confirm that the bootstrapper.json has different content because of change to the `VSTargetChannel` here.

Given this information, I think the only way to unblock private builds from merging is to queue a new build manually setting `VSTargetChannelOverride` to `int.d.17.3` so that it won't impact official builds. I am happy to close this PR because team has a temporary workaround.

Reply:
We can consider adding a separate property that'd only affect what version dartlab gets. 

I agree that we can't merge it as is.

Reply:
That property would obviously inherit from VSTargetChannel.

Reply:
When running DartLab tests, we use `bootstrapper.json` file that is output of a MicroBuild task that produces the bootstrapper.json file. Hence, I added a new task that generates bootstrapper for `int.d17.3` VS target channel only for PR builds. This change will not affect official builds.

Reply:
@kartheekp-ms If we build two bootstrappers, one for Apex/E2E tests, and another for OptProf, then we need to change this script that gets the URL for E2E and Apex tests: https://github.com/NuGet/NuGet.Client/blob/451713c59f5206f77c97d093dd5191be13f9d0e0/eng/pipelines/templates/Build_and_UnitTest.yml#L317-L332

We'll also need to figure out how to modify the OptProf pipeline to get the correct BootstrapperURL: https://github.com/NuGet/NuGet.Client/blob/451713c59f5206f77c97d093dd5191be13f9d0e0/eng/pipelines/optprof.yml#L125-L129

Maybe it's worthwhile having the OptProf pipeline build its own bootstrapper, that way the build pipeline only builds one (used for E2E and Apex), and we don't need to worry about OptProf re-using it.

Reply:
> If we build two bootstrappers, one for Apex/E2E tests, and another for OptProf, then we need to change this script that gets the URL for E2E and Apex tests:

We are not building two bootstrappers instead we are building different bootstrapper for PR and official builds. PR builds temporarily use `int.d17.3` channel because E2E tests are failing on `int.main` channel.

In the [PR build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6743384&view=logs&j=82de8b56-c0c9-5dfe-8c8b-fe72738a28a9&t=68f29c59-da14-5ade-fb7e-9eb80997eb62
) for this change the bootstrapper for official build didn't run because the condition evaluated to false.

![image](https://user-images.githubusercontent.com/52756182/192600330-d487e73b-9de3-4049-ad92-7aeb98688d02.png)

I am planning on reverting this change once https://developercommunity.visualstudio.com/t/VS-2022-174-preview-versions-throws-a/10157815 feedback ticket is resolved.

The purpose of this PR is to avoid the need for 2 private builds (one automatically kicked off when a commit is pushed, manually queueing a build by overriding VS target channel to unblock merging into dev). I thought it is a safe approach because official builds are not impacted, and we are not skipping the entire E2E part1 test suite.

> Maybe it's worthwhile having the OptProf pipeline build its own bootstrapper, that way the build pipeline only builds one (used for E2E and Apex), and we don't need to worry about OptProf re-using it.

Good idea. I am happy to create a follow-up issue for this comment in our `Engineering Productivity` backlog and work on it next sprint if that's okay.

Reply:
Can we pass a different property instead of declaring the bootstrapper task multiple times?